File: epadel_mobile/lib/models/rezervacije.dart
import 'package:epadel_mobile/models/models.dart';
import 'package:json_annotation/json_annotation.dart';

part 'rezervacije.g.dart';

@JsonSerializable()
class Rezervacija {
  int? korisnikId;
  int? terenId;
  String? terenNaziv;
  String? vrijemeRezervacije;
  String? datumRezervacije;
  String? paymentMethod;
  String? potrebnaReketa;
  String? korisnickoIme;
  String? lokacija;
  int? cijena;
  int? brojReketa;
  Teren? teren;

  Rezervacija({
    this.terenNaziv,
    this.korisnickoIme,
    this.lokacija,
    this.cijena,
    this.brojReketa,
    this.teren,
  });

  factory Rezervacija.fromJson(Map<String, dynamic> json) =>
      _$RezervacijaFromJson(json);
  Map<String, dynamic> toJson() => _$RezervacijaToJson(this);
}

File: ePadel/ePadel.Model/Rezervacije.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ePadel.Model
{
    public class Rezervacije
    {
        public int RezervacijaId { get; set; }
        public int? KorisnikId { get; set; }
        public int? TerenId { get; set; }
        public int? Cijena { get; set; }
        public string? RezervacijaStatus { get; set; }
        public string? DatumRezervacije { get; set; }
        public string? VrijemeRezervacije { get; set; }
        public string? PotrebnaReketa { get; set; }
        public int? BrojReketa { get; set; }
        //public string PaymentMethod
        //{
        //    get { return PlatiTermin.PaymentMethod; }
        //}
        public string? TerenNaziv
        {
            get
            {
                return Teren?.Naziv;
            }
        }
        public string? KorisnickoIme
        {
            get
            {
                return Korisnik?.KorisnickoIme;
            }
        }
        public virtual Korisnik? Korisnik { get; set; }
        public virtual Tereni? Teren { get; set; }
        public virtual PlatiTermin? PlatiTermin { get; set; }


        //public virtual ICollection<Plaćanja>? Plaćanjas { get; set; }
    }
}
